dataloader.go

```diff
@@ -277,14 +277,17 @@ func (l *Loader) LoadMany(originalContext context.Context, keys []string) ThunkM
 	wg := sync.WaitGroup{}
 
 	wg.Add(length)
-	for i := range keys {
-		go func(ctx context.Context, i int) {
+	for i, key := range keys {
+		// Request the key serially to ensure the result order is maintained in key order.
+		thunk := l.Load(ctx, key)
```

Comment:
Do we need this change on this version? If the index is being passed to the `go func` then the order of the results should be correct. Am I missing something?

Reply:
That said, I don't think this change hurts anything so I'll approve.

Reply:
It might not be needed in this version, I might've got the cache Clear issues crossed with the non-deterministic LoadMany issue from the earlier versions.